arm-operationalinsights/2015-03-20/swagger/OperationalInsights.json

```diff
@@ -0,0 +1,1110 @@
+        },
+        "start": {
+          "type": "string",
+          "format": "date-time",
+          "description": "The start date filter, so the only query results returned are after this date."
+        },
+        "end": {
+          "type": "string",
+          "format": "date-time",
+          "description": "The end date filter, so the only query results returned are before this date."
+        }
+      },
+      "description": "Parameters specifying the search query and range."
+    },
+    "StorageAccount": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "The Azure Resource Manager ID of the storage account resource."
+        },
+        "key": {
+          "type": "string",
+          "description": "The storage account key."
+        }
+      },
+      "description": "Describes a storage account connection."
+    },
+    "StorageInsightStatus": {
+      "properties": {
+        "state": {
+          "type": "string",
+          "description": "The state of the storage insight connection to the workspace"
+        }
+      },
+      "description": "The status of the storage insight."
+    },
+    "StorageInsightProperties": {
+      "properties": {
+        "containers": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The names of the blob containers that the workspace should read"
+        },
+        "tables": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The names of the Azure tables that the workspace should read"
+        },
+        "storageAccount": {
+          "$ref": "#/definitions/StorageAccount",
+          "description": "The storage account connection details"
+        },
+        "status": {
+          "$ref": "#/definitions/StorageInsightStatus",
+          "description": "The status of the storage insight"
+        }
+      },
+      "required": [
+        "storageAccount"
+      ],
+      "description": "Storage insight properties."
+    },
+    "StorageInsight": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/StorageInsightProperties",
+          "description": "Storage insight properties."
+        },
+        "eTag": {
+          "type": "string",
+          "description": "The ETag of the storage insight."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "description": "The top level storage insight resource container."
+    },
+    "StorageInsightListResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/StorageInsight"
+          },
+          "description": "Gets or sets a list of storage insight instances."
+        },
+        "@odata.nextLink": {
```

Comment:
you can use the extension `"x-ms-client-name": "odataNextLink"` over here. This would be the name of the property in the model class. The serialization name would still remain "@odata.nextLink".